app/src/views/SingleTeamView.vue

```diff
@@ -73,69 +74,101 @@
         @updateTeam="() => updateTeamAPI()"
       />
     </ModalComponent>
-    <CreateBankModal
-      v-if="bankModal"
-      @close-modal="() => (bankModal = false)"
-      @create-bank="async () => deployBankContract()"
-      :loading="createBankLoading"
-    />
-    <DepositBankModal
-      v-if="depositModal"
-      @close-modal="() => (depositModal = false)"
-      @deposit="async (amount: string) => depositToBank(amount)"
-      :loading="depositLoading"
-    />
+    <ModalComponent v-model="bankModal">
+      <CreateBankForm
+        @close-modal="() => (bankModal = false)"
+        @create-bank="async () => deployBankContract()"
+        :loading="createBankLoading"
+      />
+    </ModalComponent>
+
+    <ModalComponent v-model="depositModal">
+      <DepositBankForm
+        v-if="depositModal"
+        @close-modal="() => (depositModal = false)"
+        @deposit="async (amount: string) => depositToBank(amount)"
+        :loading="depositLoading"
+      />
+    </ModalComponent>
     <TransferFromBankModal
       v-if="transferModal"
       @close-modal="() => (transferModal = false)"
       @transfer="async (to: string, amount: string) => transferFromBank(to, amount)"
-      @searchMembers="async (query: string) => await searchMembers(query)"
-      :filteredMembers="filteredMembers"
+      @searchMembers="(input) => searchUsers({ address: input, name: '' })"
+      :filteredMembers="foundUsers"
       :loading="transferLoading"
```

Comment:
Instead of creating many modal, why not creating one modal and decide what should appear inside ?

Reply:
That's what I have done based on the structure u provided (haven't implemented for this since the PR was getting too long). I will implement it for all remaining modals in the next PR.

Reply:
Nice